lib/elliptic/ec/index.js

```diff
@@ -164,3 +169,42 @@ EC.prototype.verify = function verify(msg, signature, key, enc) {
 
   return p.getX().mod(this.n).cmp(r) === 0;
 };
+
+EC.prototype.recoverPubKey = function(msg, signature, j) {
+  var n = this.n;
+  var g = this.g;
+
+  var r = signature.r;
+  var s = signature.s;
+
+  // A set LSB signifies that the y-coordinate is odd
+  var isYOdd = j & 1;
+
+  // The more significant bit specifies whether we should use the
+  // first or second candidate key.
+  var isSecondKey = j >> 1;
+
+  //1.1. Let x = r + jn.
+  var x = isSecondKey ? n.add(r) : r;
+  // console.log('x: ' + x.toString());
+
+  var r = this.curve.pointFromX(isYOdd, x);
+
+  //mul doesn't work
+  var rn = r.mul(n);
```

Comment:
@indutny there appears to be a problem with Point.mul here. some points on the curver appear to not be in red when they should.  Using this tests case https://gist.github.com/wanderer/d1726fea77e5f2a8d46d